test-plan/eventing/subscription-lifecycle-conformance.md

```diff
@@ -0,0 +1,257 @@
+```
+
+Tested in eventing:
+- 
+
+### [Output]
+
+```
+{
+  "test": "control-plane/subscription-lifecycle/immutability"
+  "output": {
+	  "expectedError": "<EXPECTED_ERROR>"
+  }
+}
+```
+
+## [Test] Subscription Readiness 
+
+Check for condition type `Ready` with status `False` cause the some of the preconditions on the spec are not met: 
+
+```
+ kubectl get subscription conformance-subscription -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].status}"
+```
+
+Lets now check the reason why the Subscription is not ready yet:
+
+```
+ kubectl get subscription conformance-subscription -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].reason}"
+```
+
+Tested in eventing:
+- 
+
+### [Output]
+
+```
+{
+  "test": "control-plane/subscription-lifecycle/subscription-readiness"
+  "output": {
+    "expectedType": "Ready",
+    "expectedStatus": "Unknown",
+    "expectedReason: "ChannelReferenceFailed"
+  }
+}
+```
+
+## [Pre] Creating the Channel
+
+```
+kubectl apply -f control-plane/subscription-lifecycle/channel.yaml
+```
+
+## [Test] Subscription Readiness 2
+
+Check for condition type `Ready` with status `False` cause even with the Channel some of the preconditions on the spec are not met: 
+
+```
+ kubectl get subscription conformance-subscription -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].status}"
+```
+
+Lets now check the reason why the Subscription is not ready yet:
+
+```
+ kubectl get subscription conformance-subscription -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].reason}"
+```
+
+Tested in eventing:
+- 
+
+### [Output]
+
+```
+{
+  "test": "control-plane/subscription-lifecycle/subscription-readiness-2"
+  "output": {
+	  "expectedType": "Ready",
+	  "expectedStatus": "False",
+    "expectedReason: "SubscriberResolveFailed"
+  }
+}
+```
+
+## [Pre] Creating the DeadLetterSink and the Subscriber
+
+```
+kubectl apply -f control-plane/subscription-lifecycle/services.yaml
+```
+
+## [Test] Subscription status update
+
+The `status.physicalSubscription` should have been updated with the 
+
+```
+kubectl get subscription conformance-subscription -ojsonpath="{.status.physicalSubscription}"
+```
+
+Tested in eventing:
+- 
+
+### [Output]
+
+```
+{
+  "test": "control-plane/subscription-lifecycle/subscription-status-update"
+  "output": {
+	  "obtainedPhysicalSubscription": {
+      "deadLetterSinkUri": "<DEADLETTERSINK_URI>",
+      "subscriberUri": "<SUBSCRIBER_URI>"
+    }
+  }
+}
+```
+
+## [Test] Subscription Readiness 2
+
+Check for condition type `Ready` with status `False` cause with even with the Channel some of the preconditions on the spec are not met: 
+
+```
+ kubectl get subscription conformance-subscription -ojsonpath="{.status.conditions[?(@.type == \"Ready\")].status}"
+```
+
+Tested in eventing:
+- 
+
+### [Output]
+
+```
+{
+  "test": "control-plane/subscription-lifecycle/subscription-readiness-2"
+  "output": {
+	  "expectedType": "Ready",
+	  "expectedStatus": "True"
+  }
+}
+```
```

Comment:
I am not sure what is being tested here. The subscription is now Ready and I guess that this test is about that but the name of the section ([Test] Subscription Readiness 2) and the first paragraph does not clarify it.

Can those be rephrased?